runtime/include/module.h

```diff
@@ -141,12 +158,62 @@ module_release(struct module *module)
 	return;
 }
 
+static inline struct wasm_stack *
+module_allocate_stack(struct module *self)
+{
+	assert(self != NULL);
+
+	struct wasm_stack *stack = wasm_stack_pool_remove_nolock(&self->pools.stack[worker_thread_idx]);
+
+	if (stack == NULL) {
+		stack = wasm_stack_new(self->stack_size);
+		if (unlikely(stack == NULL)) return NULL;
+	}
+
+	return stack;
+}
+
+static inline void
+module_free_stack(struct module *self, struct wasm_stack *stack)
+{
+	wasm_stack_reinit(stack);
+	wasm_stack_pool_add_nolock(&self->pools.stack[worker_thread_idx], stack);
+}
+
+static inline struct wasm_memory *
+module_allocate_linear_memory(struct module *module)
+{
+	assert(module != NULL);
+
+
+	size_t initial = (size_t)module->abi.starting_pages * WASM_PAGE_SIZE;
+	size_t max     = (size_t)module->abi.max_pages * WASM_PAGE_SIZE;
+
+	assert(initial <= (size_t)UINT32_MAX + 1);
```

Comment:
Are we bound to only 64 bit architectures? I don't know if we want to make implicit assumptions about only working on 64 bit architectures, though I doubt we'll ever run on 32.

I think that there is a cleaner way to express this without requiring size_t > int.

Reply:
thinking about this a little more, I noticed the `=` here. Not sure why we would allow it to be over the max.

`assert(initial & (UINT32_MAX - 1) == 0)` ?

assumes UINT32_MAX is a power of 2, and 2s complement. I'm OK with tehse assumptions. Up to you.